backend/src/payment/providers/checkout-v3.ts

```diff
@@ -0,0 +1,209 @@
+          log.error('Response signature validation failed', {order_id: order.order_id});
+    }
+    log.error('Posting new payment failed', {error: {data: err.response.data, status: err.response.status, header: err.response.headers}, provider: PROVIDER});
+    return reject(err.response.data);
+  }
+
+  if (!verify(resp.headers.signature, resp.headers, resp.data)) {
+    return reject({name: 'Verification error', message: 'Signature verification failed'});
+  }
+
+  const createResp: payment.ICreateResponse = {
+    payment_id: resp.data.transactionId,
+    redirect_url: resp.data.href,
+    payment_url: resp.data.href,
+    payment_provider: PROVIDER,
+    payload: resp.data.providers
+  }
+  
+  return resolve(createResp);
+}
+
+// TODO: implement callback handlers
+export function handleSuccessCallback(req: express.Request): Promise<payment.ISuccessResponse> {
+  const headers = req.query;
+
+  if (! verify(headers.signature, headers)) {
+    return reject({name: 'Verification error', message: 'Signature verification failed'});
+  }
+
+  return resolve({
+    payment_id: headers['checkout-transaction-id'],
+    payment_provider: PROVIDER,
+  });
+}
+
+export function handleCancelCallback(req: express.Request): Promise<payment.ICancelResponse> {
+  const headers = req.query;
+
+  if (! verify(headers.signature, headers)) {
+    return reject({name: 'Verification error', message: 'Signature verification failed'});
+  }
+
+  return resolve({
+    payment_id: headers['checkout-transaction-id'],
+    payment_provider: PROVIDER,
+  });
+}
+
+export function checkStatus(payment_id: string, payment_url: string): Promise<payment.IStatusResponse> {
+
+
+  return resolve({
+    payment_id,
+    payment_url,
+    status: 'not-implemented',
+  });
+}
+
+function sign(headers: {[key: string]: any}, body?: CreateRequestBody): string {
+  var payloadArray =
+    Object.keys(headers)
+      .filter((value: string) => {
+        return /^checkout-/.test(value);
+      })
+      .sort()
+      .map((key) => [ key, headers[key] ].join(':'))
+
+  var payload: string;
+  if(body) payload = payloadArray.concat(JSON.stringify(body)).join("\n");
+  else payload = payloadArray.concat('').join("\n");
+
+  var hmac = crypto
+    .createHmac('sha256', config.password)
+    .update(payload)
+    .digest('hex');
+
+    return hmac;
+  }
+
+function verify(expectedSignature: string, headers: {[key: string]: any}, body?: CreateRequestBody): boolean {
+  var signature = sign(headers, body);
+  return signature === expectedSignature
+}
+
+function ticketToRequestBodyItem(ticket: ticket.ITicket): RequestBodyItem {
+  
+  var item: RequestBodyItem = {
+    unitPrice: ticket.ticket_price * 100,
+    units: 1,
+    vatPercentage: 0,
+    productCode: ticket.production_title + '-' + ticket.show_title + '-' + ticket.seat_number,
+    deliveryDate: moment().format('YYYY-MM-DD')
+  }
+
+  return item;
```

Comment:
I'd just create this inline again, no need to create the var

Reply:
Same applies for all similar cases, no need to create a variable if it's not used anywhere  - the function name and return type should be indicative enough. (Unless there's some typescript restrictions here that require specifying the var to get its type?)

Reply:
I'm mostly using this syntax to typecheck my own implementation.

Reply:
But as you point out, this can be removed.